matrix_sdk/src/client.rs

```diff
@@ -1407,7 +1407,7 @@ impl Client {
     #[instrument]
     async fn claim_one_time_keys(
         &self,
-        one_time_keys: BTreeMap<UserId, BTreeMap<DeviceId, KeyAlgorithm>>,
+        one_time_keys: BTreeMap<UserId, BTreeMap<Box<DeviceId>, KeyAlgorithm>>,
```

Comment:
I feel like using `DeviceId` isn't as nice as it used to be now that `DeviceId` is a type alias over `str` instead of `String`. I think we only had a couple of friction points where we wanted to use `&DeviceId`, now we have plenty of places where we need to box the device id.

I know this isn't really important for our public API since we can hide it, but investing some time into a wrapper type for this might make Ruma nicer in general for everyone.

Perhaps wrapping a `Cow<str>` might work, or the `unicase` case (🥁) mentioned over [here](https://users.rust-lang.org/t/how-to-newtype-string/5211/2) sounds nice as well.

@jplatte any thoughts on this?

Reply:
I'll open a issue over at Ruma for this since we're going to merge this.